### PR TITLE
Make avow throw POJOs

### PR DESCRIPTION
**Pull request by gunar**
Node.js's `assert` can't throw Plain Old Javascript Objects as `message`.
Well, it can but `message` is stringified into `[object Object]`.

Previously, to find the `message` argument, we were checking if the last argument was an instance of `Error`.
Now, we're checking [each specific method of assert](https://nodejs.org/api/assert.html) for its argument order.

### CODE REVIEW

**Comment by gunar** [comment]
Todo: Publish after merge.

File: index.js [hunk 1]
```diff
@@ -2,7 +2,40 @@
...
+  notEqual: 2,
+  // assert.notStrictEqual(actual, expected[, message])
+  notStrictEqual: 2,
+  // assert.ok(value[, message])
+  ok: 1,
+  // assert.strictEqual(actual, expected[, message])
+  strictEqual: 2,
+  // assert.throws(block[, error][, message])
+  throws: 2,
+}
```
Comment (keenahn) [inline index.js]: Hm, this is not great but I understand why. Could we make it a bit more DRY by using `function-parser` and checking for the `message` param?

File: index.js [hunk 2]
```diff
@@ -19,18 +52,19 @@ function avowify(fn) {
...
+      if (message) throw message
       throw e
     }
   }
 }
 
 const handler = {
   // e.g. avow.equal(...)
-  get: (target, method) => avowify(target[method]),
+  get: (target, methodName) => avowify(target[methodName], methodName),
```
Comment (keenahn) [inline index.js]: 👍

File: index.js [hunk 3]
```diff
@@ -19,18 +52,19 @@ function avowify(fn) {
...
   // e.g. avow(...)
-  apply: (target, thisArg, args) => avowify(target).apply(thisArg, args)
+  apply: (target, thisArg, args) => avowify(target, undefined).apply(thisArg, args)
```
Comment (keenahn) [inline index.js]: why, just to be more explicit? ok

File: test/index.spec.js [hunk 4]
```diff
@@ -36,16 +36,48 @@ test('Compatible behavior with assert', t => {
...
+
+test('POJO as message', t => {
+  const pojo = { key: 'value' }
+  try { avow(false, pojo) }
+  catch (e) {
+    assert.deepStrictEqual(e, pojo)
+    return t.pass()
+  }
+  t.fail()
+})
```
Comment (keenahn) [inline test/index.spec.js]: coo

**Comment by keenahn** [comment]
@gunar Looks good. I'll leave it to you if you want to use `function-parser` or not.

**Pull request merged**